### PR TITLE
Fix find by name accepting any input

### PR DESCRIPTION
**Pull request by Domokunx**
Checks for invalid/impossible names

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java [hunk 1]
```diff
@@ -30,7 +31,13 @@ public FindCommand parse(String args) throws ParseException {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
+
+        // Verify nameKeywords are valid Name inputs
         List<String> nameKeywords = Arrays.asList(nameKeyArg.split("\\s+"));
+        if (!nameKeyArg.isEmpty() && !NameContainsKeywordsPredicate.areValidNameKeywords(nameKeywords)) {
```
Comment (yockcheng) [inline src/main/java/seedu/address/logic/parser/FindCommandParser.java:37]: Should we handle this in NameContainsKeywordsPredicate constructor instead?

**Review by yockcheng — COMMENTED** [review]
Personally, I feel that it makes more sense to handle the checking in the constructor of NameContainsKeywordsPredicate then either returning the object successfully or throwing an Exception.

File: src/main/java/seedu/address/logic/parser/FindCommandParser.java (hunk 1 above)
Comment (Domokunx) [reply src/main/java/seedu/address/logic/parser/FindCommandParser.java]: Refactored to use ParserUtil instead to preserve SLAP

**Review by DKHB0207 — APPROVED** [review]
LGTM!

**Pull request merged**